SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalanceEnforcer/SpatialLoadBalanceEnforcerTest.cpp

```diff
@@ -0,0 +1,262 @@
+{
+	TUniquePtr<SpatialVirtualWorkerTranslator>  VirtualWorkerTranslator = CreateVirtualWorkerTranslator();
+
+	// Here we create a static component view and add entities to it but do not assign the AuthorityIntent virtual worker id.
+	// This means that the load balance enforcer will not be able to find the physical worker id associated with an entity and therefore fail to produce ACL requests.
+	USpatialStaticComponentView* StaticComponentView = NewObject<USpatialStaticComponentView>();
+	AddEntityToStaticComponentView(*StaticComponentView, EntityIdOne, SpatialConstants::INVALID_VIRTUAL_WORKER_ID, WORKER_AUTHORITY_AUTHORITATIVE);
+	AddEntityToStaticComponentView(*StaticComponentView, EntityIdTwo, SpatialConstants::INVALID_VIRTUAL_WORKER_ID, WORKER_AUTHORITY_AUTHORITATIVE);
+
+	TUniquePtr<SpatialLoadBalanceEnforcer> LoadBalanceEnforcer = MakeUnique<SpatialLoadBalanceEnforcer>(ValidWorkerOne, StaticComponentView, VirtualWorkerTranslator.Get());
+
+	LoadBalanceEnforcer->QueueAclAssignmentRequest(EntityIdOne);
+	LoadBalanceEnforcer->QueueAclAssignmentRequest(EntityIdTwo);
+
+	TArray<SpatialLoadBalanceEnforcer::AclWriteAuthorityRequest> ACLRequests = LoadBalanceEnforcer->ProcessQueuedAclAssignmentRequests();
+
+	bool bSuccess = ACLRequests.Num() == 0;
+	TestTrue("LoadBalanceEnforcer returned expected ACL assignment results", bSuccess);
+
+	return true;
+}
+
+LOADBALANCEENFORCER_TEST(GIVEN_load_balance_enforcer_with_valid_mapping_WHEN_asked_for_acl_assignments_THEN_return_correct_acl_assignment_requests)
+{
+	TUniquePtr<SpatialVirtualWorkerTranslator>  VirtualWorkerTranslator = CreateVirtualWorkerTranslator();
+
+	USpatialStaticComponentView* StaticComponentView = NewObject<USpatialStaticComponentView>();
+	AddEntityToStaticComponentView(*StaticComponentView, EntityIdOne, VirtualWorkerOne, WORKER_AUTHORITY_AUTHORITATIVE);
+	AddEntityToStaticComponentView(*StaticComponentView, EntityIdTwo, VirtualWorkerTwo, WORKER_AUTHORITY_AUTHORITATIVE);
+
+	TUniquePtr<SpatialLoadBalanceEnforcer> LoadBalanceEnforcer = MakeUnique<SpatialLoadBalanceEnforcer>(ValidWorkerOne, StaticComponentView, VirtualWorkerTranslator.Get());
+
+	LoadBalanceEnforcer->QueueAclAssignmentRequest(EntityIdOne);
+	LoadBalanceEnforcer->QueueAclAssignmentRequest(EntityIdTwo);
+
+	TArray<SpatialLoadBalanceEnforcer::AclWriteAuthorityRequest> ACLRequests = LoadBalanceEnforcer->ProcessQueuedAclAssignmentRequests();
+
+	bool bSuccess = true;
+	if (ACLRequests.Num() == 2)
+	{
+		bSuccess &= ACLRequests[0].EntityId == EntityIdOne;
+		bSuccess &= ACLRequests[0].OwningWorkerId == ValidWorkerOne;
+		bSuccess &= ACLRequests[1].EntityId == EntityIdTwo;
+		bSuccess &= ACLRequests[1].OwningWorkerId == ValidWorkerTwo;
+	}
+	else
+	{
+		bSuccess = false;
+	}
+
+	TestTrue("LoadBalanceEnforcer returned expected ACL assignment results", bSuccess);
+
+	return true;
+}
```

Comment:
Can you also test that:

	LoadBalanceEnforcer->QueueAclAssignmentRequest(EntityIdOne);
	LoadBalanceEnforcer->QueueAclAssignmentRequest(EntityIdOne);

results in only a single ACLrequest?

Reply:
Ahh this is actually a mistake in my last commit, it should read EntityIdTwo on the second line. That said that would be another nice little test :)